FirebasePerformance/Sources/FPRNanoPbUtils.h

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <TargetConditionals.h>
+#if __has_include("CoreTelephony/CTTelephonyNetworkInfo.h") && !TARGET_OS_MACCATALYST
+#define TARGET_HAS_MOBILE_CONNECTIVITY
+#import <CoreTelephony/CTTelephonyNetworkInfo.h>
+#endif
+
+#import "FirebasePerformance/Sources/AppActivity/FPRTraceBackgroundActivityTracker.h"
+#import "FirebasePerformance/Sources/Instrumentation/FPRNetworkTrace.h"
+#import "FirebasePerformance/Sources/Public/FIRTrace.h"
+
+#import "FirebasePerformance/Sources/Protogen/nanopb/perf_metric.nanopb.h"
+
+typedef struct {
+  pb_bytes_array_t* _Nonnull key;
+  pb_bytes_array_t* _Nonnull value;
+} StringToStringMap;
+
+typedef struct {
+  pb_bytes_array_t* _Nonnull key;
+  bool has_value;
+  int64_t value;
+} StringToNumberMap;
+
+extern pb_bytes_array_t* _Nullable FPREncodeData(NSData* _Nonnull data);
+
+extern pb_bytes_array_t* _Nullable FPREncodeString(NSString* _Nonnull string);
+
+extern NSData* _Nullable FPRDecodeData(pb_bytes_array_t* _Nonnull pbData);
+
+extern NSString* _Nullable FPRDecodeString(pb_bytes_array_t* _Nonnull pbData);
```

Comment:
Would appreciate some comments for each of these functions for better reference lookup.

Reply:
May be even consider something like FPRNanoPbDecodeUtils?

Reply:
Comments are added.